src/api/goods.ts

```diff
@@ -1,11 +1,34 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL)
-    .then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`${response.status} - ${response.statusText}`);
+    }
+
+    const goods = await response.json();
+
+    return goods;
+  } catch (error) {
+    throw new Error(`The problem is here ${error}`);
+  }
 }
 
-export const get5First = () => {};
+export const get5First = async (): Promise<Good[]> => {
+  const data = await getAll();
+   return [...data].sort((product1, product2) => {
+    return product1.name.localeCompare(product2.name);
+  }).slice(0, 5);
+
+  
+};
```

Comment:
Chained methods should have same indentation.
```suggestion
   return [...data]
     .sort((product1, product2) => {
       return product1.name.localeCompare(product2.name);
     })
     .slice(0, 5);
};
```